ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -1440,64 +1441,112 @@ template<class FluidSystem,class Scalar>
 void EclGenericOutputBlackoilModule<FluidSystem,Scalar>::
 outputErrorLog(const Comm& comm) const
 {
-    const size_t maxNumCellsFaillog = 20;
+    const auto pbSize = static_cast<int>(this->failedCellsPb_.size());
+    const auto pdSize = static_cast<int>(this->failedCellsPd_.size());
```

Comment:
While changing this anyway, maybe we should resort to an utiility function that I implemented for doing this to reduce code:

```
#include<opm/grid/common/CommunicationUtils.hpp>
....
const auto int root = 0;
auto [ globalFailedCellsPb, displPb] = gatherv(this->failedCellsPb_, root);
auto [ globalFailedCellsPd, displPd] = gatherv(this->failedCellsPd_, root);
....
```

This functionality is of course too well hidden in opm-grid...

Reply:
That's a good tip.  It does have the same undefined behaviour of [forming](https://github.com/OPM/opm-grid/blob/fa96a96fa2a8bc89d74985ff5a2203a99841ef87/opm/grid/common/CommunicationUtils.hpp#L87) `displ.begin() + 1` on non-root processes as this code used to have, but I suppose we can fix it there too.

Reply:
Yep, you are right. It is undefined behavior, also the documentation is not 100% crystal clear (iterator must be in derefenceable state before increment.

Reply:
> Maybe use displ.begin() + (rank==root) or displ.begin() + (displ.begin() != displ.end()), then.

Or just not run the `partial_sum()` on non-root processes since by construction `displ` is empty on those.  I'd probably implement `gatherv()` along the lines of
```C++
// ...
comm.gather(&mySize, sizes.data(), 1, root);
std::vector<T,A> output;
if (isRoot) {
    std::partial_sum(sizes.begin(), sizes.end(), displ.begin() + 1);
    output.resize(displ.back(), T{});
}
comm.gatherv(...);
return { output, displ };
```